docs/book/developer-guide/advanced-concepts/materializer.md

```diff
@@ -0,0 +1,454 @@
+---
+description: How to use materializers to pass custom data types through steps.
```

Comment:
Same as above, I don't think materializers really version your data, they just save it. Versioning happens implicitly by storing artifacts of separate runs in separate locations and having a metadata store that tracks it, no? Personally I would rather elevate the metadata store as the component that does the data versioning since it is a well-established ML component and a core concept of every ZenML stack (unlike materializers, which some users might never have to interact with).